timesketch/api/v1/resources/sigma.py

```diff
@@ -153,11 +217,371 @@ def post(self):
+                    rule_uuid, parsed_rule.get("id")
+                ),
+            )
+
+        sigma_rule_from_db = SigmaRule.query.filter_by(
+            rule_uuid=rule_uuid
+        ).first()
+
+        if not sigma_rule_from_db:
+            error_msg = "Sigma rule with uuid: {0!s} not found".format(
+                rule_uuid
+            )
+            logger.error(error_msg)
+            abort(HTTP_STATUS_CODE_NOT_FOUND, error_msg)
+
+        sigma_rule_from_db.rule_yaml = rule_yaml
+        sigma_rule_from_db.title = parsed_rule.get("title")
+        sigma_rule_from_db.description = parsed_rule.get("description")
+
+        try:
+            db_session.add(sigma_rule_from_db)
+            db_session.commit()
+        except IntegrityError as e:
+            error_msg = "Error adding Sigma rule {0!s}".format(e)
+            logger.error(error_msg)
+            abort(
+                HTTP_STATUS_CODE_CONFLICT,
+                error_msg,
+            )
+
+        return self.to_json(
+            sigma_rule_from_db, status_code=HTTP_STATUS_CODE_OK
+        )
+
+
+class SigmaRuleByTextResource(resources.ResourceMixin, Resource):
+    """Get a Sigma rule by providing a text."""
+
+    @login_required
+    def post(self):
+        """Get a parsed Sigma rule by providing a text
+
+        Will parse a provided text `rule_yaml`, parse it and return as JSON
+
+
+        If no form content is given, the method will abort with HTTP Error 400.
+
+        Returns:
+            JSON sigma rule e.g.
+                {"objects": [sigma_rule], "meta": {"parsed": True}}
+        """
+
+        form = request.json
+        if not form:
+            form = request.data
+
+        content = form.get("content")
+        if not content:
+            return abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Missing value in the request. Nothing to parse",
+            )
+
+        try:
+            sigma_rule = ts_sigma_lib.parse_sigma_rule_by_text(content)
+
+        except ValueError as e:
+            error_msg = (
+                "Sigma rule Parsing error with provided rule {0!s}".format(e)
+            )
+            logger.error(
+                error_msg,
+                exc_info=True,
+            )
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                error_msg,
+            )
+
+        except NotImplementedError as e:
+            error_msg = (
+                "Sigma Parsing error: Feature in the rule provided"
+                " is not implemented in this backend {0!s}".format(e)
+            )
+            logger.error(
+                error_msg,
+                exc_info=True,
+            )
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                error_msg,
+            )
+
+        except sigma_exceptions.SigmaParseError as e:
+            error_msg = "Sigma parsing error generating rule"
+            " with error: {0!s}".format(e)
```

Comment:
```suggestion
            error_msg = "Sigma parsing error: {0!s}".format(e)
```

`e` needs to be coerced as it's an Exception and not a string.